ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java

```diff
@@ -180,7 +181,7 @@ private void appendIndex(QueryEntity queryEntity, AssociationKeyMetadata associa
 			fields.put( realColumnName, true );
 		}
 		queryIndex.setFields( fields );
-		queryIndex.setName( queryEntity.getTableName() + '_' + org.hibernate.ogm.util.impl.StringHelper.join( fields.keySet(), "_" ) );
+		queryIndex.setName( UUID.randomUUID().toString().replace( "-", "_" ) );
```

Comment:
A random name will work but it won't be very user friendly, making it hard to debug when an error occurs or reproduce problems.
Let's use the previous approach and only if the string is bigger than 255 chars do something different. What happens if index name is bigger than 255 chars? Error?